### PR TITLE
Fix #5866: plot legend

### LINKED ISSUE #5866: don't reset plot legend selection on load()
In the accel app, when new data is loaded into a plot, the y legend resets to "all selected". Instead, it should maintain the current visible/hidden selection unless the y1/y2/y3 field names have changed.

So the use case would be:
- you are viewing a parameter plot with multiple y values
- you hide one of the values from the legend
- the plot reloads (either automatically refreshing (accel), or you select a different frame (elegant)
- the hidden plot value should remain hidden

Or the alternative case where the legend should be reset:
- you view a parameter plot with some y values hidden
- you open the plot settings and choose a different set of y values
- now the plot should reset showing all values (don't preserve hidden/visible status)

To reproduce, open the elegant "bunchCompLSC - lattic" example, and in the B1 output panel, add multiple y values. Hide a value, then page through the frames and notice the legend gets reset.

x

### PR DESCRIPTION
**Pull request by gurhar1133**
cache legend toggled off/on

### CODE REVIEW

File: sirepo/package_data/static/js/sirepo-plotting.js [hunk 1]
```diff
@@ -3541,6 +3540,16 @@ SIREPO.app.directive('parameterPlot', function(appState, focusPointService, layo
                 return true;
             }
 
+            function cachedToggleOff(pIndex, modelName) {
+                appState.models.toggledOff = appState.models.toggledOff || {};
+                appState.models.toggledOff[modelName] = appState.models.toggledOff[modelName] || {};
+                if (!appState.models.toggledOff[modelName].hasOwnProperty(pIndex)) {
+                  appState.models.toggledOff[modelName][pIndex] = false;
+                  return false;
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-plotting.js:3548]: You're returning the value below anyway, so you can omit this return

File: sirepo/package_data/static/js/sirepo-plotting.js [hunk 2]
```diff
@@ -3690,9 +3699,13 @@ SIREPO.app.directive('parameterPlot', function(appState, focusPointService, layo
                 });
             }
 
-            function togglePlot(pIndex) {
+            function togglePlot(pIndex, modelName) {
                 setPlotVisible(pIndex, ! isPlotVisible(pIndex));
                 updateYLabel();
+                if (appState.models.toggledOff) {
+                    appState.models.toggledOff[modelName][pIndex] = !appState.models.toggledOff[modelName][pIndex];
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-plotting.js:3706]: Space after bang

File: sirepo/package_data/static/js/sirepo-plotting.js [hunk 3]
```diff
@@ -3690,9 +3699,13 @@ SIREPO.app.directive('parameterPlot', function(appState, focusPointService, layo
...
+                }
+                appState.saveChanges("toggledOff");
```
Comment (mkeilman) [inline sirepo/package_data/static/js/sirepo-plotting.js:3708]: Single quotes

**Review by mkeilman — COMMENTED** [review]
Minor comments

File: sirepo/package_data/static/js/sirepo-plotting.js [hunk 4]
```diff
@@ -4014,6 +4027,11 @@ SIREPO.app.directive('parameterPlot', function(appState, focusPointService, layo
                     setPlotVisible(ip, true);
                 });
                 updateYLabel();
+                plots.forEach(function(plot, i) {
+                    if (cachedToggleOff(i, $scope.modelName)) {
+                        setPlotVisible(i, ! isPlotVisible(i));
```
[the remark this answers is not in the document]
Comment (gurhar1133) [reply sirepo/package_data/static/js/sirepo-plotting.js]: Doesn't work

[the remark this answers is not in the document]
Comment (mkeilman) [reply sirepo/package_data/static/js/sirepo-plotting.js]: OK - not worth worrying about imho

**Review by mkeilman — APPROVED** [review]
Cool!

**Comment by moellep** [comment]
This doesn't seem to solve this use-case from the issue:

Or the alternative case where the legend should be reset:

- you view a parameter plot with some y values hidden
- you open the plot settings and choose a different set of y values
- now the plot should reset showing all values (don't preserve hidden/visible status)

**Comment by gurhar1133** [comment]
> This doesn't seem to solve this use-case from the issue:
> 
> Or the alternative case where the legend should be reset:
> 
> * you view a parameter plot with some y values hidden
> * you open the plot settings and choose a different set of y values
> * now the plot should reset showing all values (don't preserve hidden/visible status)

Sorry I missed this. Working on it

**Comment by gurhar1133** [comment]
@moellep have another look. I think I fixed

**Comment by moellep** [comment]
Looks great!

**Pull request merged**